geth/transactions/queue/queue.go

```diff
@@ -36,33 +33,38 @@ var (
 	ErrInvalidCompleteTxSender = errors.New("transaction can only be completed by the same account which created it")
 )
 
+// remove from queue on any error (except for transient ones) and propagate
+var transientErrs = map[string]bool{
```

Comment:
Is there any reason why this map is not `map[error]bool`?

Reply:
it is for errors returned from an eth client,` jsonError{Message: "error"}`

Reply:
Ah ok. Errors from `EthClient` are dynamically created so even if an error has the same message, it will be a different object, thus relying on strings. Make sense 👍  Also, cool that error messages are derived from packages so everything should be fine.

I would suggest adding a comment explaining this structure as looking at its initialization it's tempting to change it to `map[error]bool`. Or a unit test that will test this property.